--- fjcontrib-1.046.orig/Makefile.in	2022-03-10 14:07:19.928044957 +0100
+++ fjcontrib-1.046/Makefile.in	2022-03-10 14:07:54.510048944 +0100
@@ -57,7 +57,7 @@
 	$(CXX) @DYNLIBOPT@ -fPIC -DPIC -Wl,-Ol -Wl,--as-needed -Wl,-soname,fastjetcontribfragile.so.0 $(CXXFLAGS) `$(FASTJETCONFIG) --cxxflags --libs` $(fragile_SHARED_SRC_LIST) -o libfastjetcontribfragile.@DYNLIBEXT@
 
 fragile-shared-install: fragile-shared
-	utils/install-sh -c -m 755 libfastjetcontribfragile.@DYNLIBEXT@ $(PREFIX)/lib
+	utils/install-sh -c -m 755 libfastjetcontribfragile.@DYNLIBEXT@ $(PREFIX)/$(get_libdir)
 	@DYNLIBPOSTPROC@ $(PREFIX)/lib/libfastjetcontribfragile.@DYNLIBEXT@ 
 
 fragile-shared-distclean:
