--- a/Makefile	2023-04-05 16:10:47.096683115 +0200
+++ b/Makefile	2023-04-05 16:10:59.461728193 +0200
@@ -155,8 +155,8 @@
 
 install: all
 	mkdir -p $(PREFIX)/bin && cp bin/* $(PREFIX)/bin/
-	mkdir -p $(PREFIX)/contrib && cp contrib/* $(PREFIX)/contrib/
-	mkdir -p $(PREFIX)/jupyter && cp jupyter/* $(PREFIX)/jupyter/
+	#mkdir -p $(PREFIX)/contrib && cp contrib/* $(PREFIX)/contrib/
+	#mkdir -p $(PREFIX)/jupyter && cp jupyter/* $(PREFIX)/jupyter/
 	mkdir -p $(PREFIX)/include && cp -r include/Professor $(PREFIX)/include/
 	test -d lib   && mkdir -p $(PREFIX)/lib   && cp -r lib/* $(PREFIX)/lib/ || true
 	test -d lib64 && mkdir -p $(PREFIX)/lib64 && cp -r lib64/* $(PREFIX)/lib64/ || true
