--- a/tools/analysis/include/base/AnalysisBase.h	2023-08-15 16:40:26.013492881 +0200
+++ b/tools/analysis/include/base/AnalysisBase.h	2023-08-15 16:40:10.846878626 +0200
@@ -21,11 +21,11 @@
 #include <TMatrixDEigen.h>
 
 #include "classes/DelphesClasses.h"
-#include "external/ExRootAnalysis/ExRootTreeReader.h"
-#include "external/ExRootAnalysis/ExRootResult.h"
+#include "ExRootAnalysis/ExRootTreeReader.h"
+#include "ExRootAnalysis/ExRootResult.h"
 
-#include "external/fastjet/JetDefinition.hh"
-#include "external/fastjet/ClusterSequence.hh"
+#include "fastjet/JetDefinition.hh"
+#include "fastjet/ClusterSequence.hh"
 
 #include "ETMiss.h"
 #include "FinalStateObject.h"

--- a/configure.ac	2023-08-15 21:52:30.431985408 +0200
+++ b/configure.ac	2023-08-15 21:52:36.640062275 +0200
@@ -26,7 +26,7 @@
 PY_PROG_PYTHON([2.7.3], ,AC_MSG_ERROR(You need Python 2.7.3+ to run CheckMATE. Use --with-python and refer to a valid python binary.) )
 AM_PATH_PYTHON([2.7.3], ,AC_MSG_ERROR(You need Python 2.7.3+ to run CheckMATE. Use --with-python and refer to a valid python binary.) )
 ROOT_PATH([5.20], , AC_MSG_ERROR(ROOT could not be initialised. Try using --with-rootsys and check whether you have ver. 5.34/06 or newer.))
-ROOT_FEATURE(python, ,AC_MSG_ERROR(ROOT could be initialised but apparently it has been configured without Python functionality. Checkout the manual to find out how to add it.))
+#ROOT_FEATURE(python, ,AC_MSG_ERROR(ROOT could be initialised but apparently it has been configured without Python functionality. Checkout the manual to find out how to add it.))
 ROOT_FEATURE(minuit2, ,AC_MSG_ERROR(ROOT could be initialised but apparently it has been configured without Minuit2 functionality. Checkout the manual to find out how to add it.))
 #ROOT_FEATURE(roofit, ,AC_MSG_ERROR(ROOT could be initialised but apparently it has been configured without RooFit functionality. Checkout the manual to find out how to add it.))
 AC_SUBST(ROOTSYS)
