diff --git a/tests/config/test_config.py b/tests/config/test_config.py
index ce691470..1b4cc234 100644
--- a/tests/config/test_config.py
+++ b/tests/config/test_config.py
@@ -8,7 +8,8 @@ from typing import TYPE_CHECKING
 
 import pytest
 
-from flatdict import FlatDict
+# Gentoo patch: flatdict dependency is unavailable
+FlatDict = pytest.importorskip('flatdict')
 
 from poetry.config.config import Config
 from poetry.config.config import boolean_normalizer
