--- a/src/Interface/OptionsHandler.hpp	2010-04-11 23:13:55.000000000 +0200
+++ b/src/Interface/OptionsHandler.hpp	2023-07-08 13:39:14.509326391 +0200
@@ -154,23 +154,23 @@
 template <class TargetType> multipleValueOption<TargetType>::multipleValueOption(const std::string& name, TargetType& valueToSet,
 		const std::string& name1, const TargetType& value1,
 		const std::string& helpString): NamedOption(name,helpString), d_valueToSet(valueToSet) {
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name1,value1));
+	d_nameMap.insert(std::make_pair(name1,value1));
 }
 template <class TargetType> multipleValueOption<TargetType>::multipleValueOption(const std::string& name, TargetType& valueToSet,
 		const std::string& name1, const TargetType& value1,
 		const std::string& name2, const TargetType& value2,
 		const std::string& helpString): NamedOption(name,helpString), d_valueToSet(valueToSet) {
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name1,value1));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name2,value2));
+	d_nameMap.insert(std::make_pair(name1,value1));
+	d_nameMap.insert(std::make_pair(name2,value2));
 }
 template <class TargetType> multipleValueOption<TargetType>::multipleValueOption(const std::string& name, TargetType& valueToSet,
 		const std::string& name1, const TargetType& value1,
 		const std::string& name2, const TargetType& value2,
 		const std::string& name3, const TargetType& value3,
 		const std::string& helpString): NamedOption(name,helpString), d_valueToSet(valueToSet) {
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name1,value1));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name2,value2));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name3,value3));
+	d_nameMap.insert(std::make_pair(name1,value1));
+	d_nameMap.insert(std::make_pair(name2,value2));
+	d_nameMap.insert(std::make_pair(name3,value3));
 }
 template <class TargetType> multipleValueOption<TargetType>::multipleValueOption(const std::string& name, TargetType& valueToSet,
 		const std::string& name1, const TargetType& value1,
@@ -178,10 +178,10 @@
 		const std::string& name3, const TargetType& value3,
 		const std::string& name4, const TargetType& value4,
 		const std::string& helpString): NamedOption(name,helpString), d_valueToSet(valueToSet) {
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name1,value1));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name2,value2));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name3,value3));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name4,value4));
+	d_nameMap.insert(std::make_pair(name1,value1));
+	d_nameMap.insert(std::make_pair(name2,value2));
+	d_nameMap.insert(std::make_pair(name3,value3));
+	d_nameMap.insert(std::make_pair(name4,value4));
 }
 template <class TargetType> multipleValueOption<TargetType>::multipleValueOption(const std::string& name, TargetType& valueToSet,
 		const std::string& name1, const TargetType& value1,
@@ -190,11 +190,11 @@
 		const std::string& name4, const TargetType& value4,
 		const std::string& name5, const TargetType& value5,
 		const std::string& helpString): NamedOption(name,helpString), d_valueToSet(valueToSet) {
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name1,value1));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name2,value2));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name3,value3));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name4,value4));
-	d_nameMap.insert(std::make_pair<std::string,TargetType>(name5,value5));
+	d_nameMap.insert(std::make_pair(name1,value1));
+	d_nameMap.insert(std::make_pair(name2,value2));
+	d_nameMap.insert(std::make_pair(name3,value3));
+	d_nameMap.insert(std::make_pair(name4,value4));
+	d_nameMap.insert(std::make_pair(name5,value5));
 }
 
 
